sherpa/instrument.py

```diff
@@ -717,10 +717,13 @@ def get_kernel(self, data, subkernel=True):
     def _check_pixel_size(self, data):
         if hasattr(self.kernel, "sky"):
             # This corresponds to the case when the kernel is actually a psf image, not just a model.
-            psf_pixel_size = self.kernel.sky.cdelt
-            data_pixel_size = data.sky.cdelt
-
-            if not numpy.allclose(psf_pixel_size, data_pixel_size):
-                warnings.warn("NOTE: The PSF pixel size ({}) does not correspond to the Image Pixel Size ({})".format(
-                    psf_pixel_size, data_pixel_size
-                ))
+            try:
+                psf_pixel_size = self.kernel.sky.cdelt
+                data_pixel_size = data.sky.cdelt
+
+                if not numpy.allclose(psf_pixel_size, data_pixel_size):
+                    warnings.warn("NOTE: The PSF pixel size ({}) does not correspond to the Image Pixel Size ({})"
+                                  .format(psf_pixel_size, data_pixel_size
+                    ))
+            except:
```

Comment:
I personally would vote for being explicit and catching only  `AttributeError`

a) I have just wasted several hours on a project of mine because a typo in a block ended up being ignored because of a bare `except`

b) I don't really buy the concept of "other things than an AttributeError could go wrong here". At the moment we are only concerned with the attribute access. Sure, the return values could do something strange in the `allclose` call, but if that happens then I'd rather we see this error, as it is a change in behavior.

Reply:
Done